### PR TITLE
fix: follow git branch naming convention based on docs.github.com

### PR DESCRIPTION
**Pull request by kmehant**
fixed based on https://docs.github.com/en/get-started/using-git/dealing-with-special-characters-in-branch-and-tag-names#naming-branches-and-tags

### CODE REVIEW

File: common/vcs/git.go
Comment (seshapad) [inline common/vcs/git.go:1]: @kmehant are these `main|development|master|(features|tests|(bug|hot)fix` the only branch prefix names? Is there any standard around this?
cc: @ashokponkumar

Comment (kmehant) [reply common/vcs/git.go:1]: The current fix makes it generic and opens up possibilities and also taking care of the breaking special characters out of branch names.

Comment (seshapad) [reply common/vcs/git.go:1]: @kmehant if you are fixing the format of the branch to specific names in the prefix which are not necessarily part of the standard, would it not affect generality?
Would it be better to let the prefix be any name which is not part of this set `main|development|master|(features|tests|(bug|hot)fix`?

File: common/vcs/git.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func isGitCommitHash(commithash string) bool {
 }
 
 func isGitBranch(branch string) bool {
-	gitBranchRegex := regexp.MustCompile(`^(main|development|master|(features|tests|(bug|hot)fix)((\/|-)[a-zA-Z0-9]+([-_][a-zA-Z0-9]+)*){1,2}|release(\/|-)[0-9]+(\.[0-9]+)*(-(alpha|beta|rc)[0-9]*)?)$`)
+	gitBranchRegex := regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9\.\-_\/]*$`)
```
Comment (kmehant) [inline common/vcs/git.go:56]: @seshapad The fix is this!

File: common/vcs/git.go
Comment (kmehant) [reply common/vcs/git.go:1]: https://github.com/konveyor/move2kube/pull/1037/files#r1209033547


> @kmehant if you are fixing the format of the branch to specific names in the prefix which are not necessarily part of the standard, would it not affect generality?


@seshapad ^ yes that's the reason we have this fix.

**Review by ashokponkumar — APPROVED** [review]

**Pull request merged**